File: snakemake/bin/scripts/hits-to-real-counts.R
# Load necessary libraries
suppressPackageStartupMessages(library(dplyr))

# Read the input file (using snakemake's input directive)
df <- read.delim(snakemake@input[[1]], sep = "\t", header = TRUE, stringsAsFactors = FALSE)

# Check if the necessary columns 'Hits' and 'Count' exist
if (!all(c('Hits', 'Count') %in% colnames(df))) {
  stop("The input file does not contain 'Hits' and/or 'Count' columns.")
}

# Create the 'RealCount' column if it does not exist
if (!"RealCount" %in% colnames(df)) {
  df$RealCount <- 0
}

# Apply the condition to update the 'RealCount' column
df <- df %>% mutate(RealCount = ifelse(Count > 0, Hits, 0))

# Save the updated dataframe to the output file (using snakemake's output directive)
write.table(df, snakemake@output[[1]], sep = "\t", row.names = FALSE, quote = FALSE)

File: snakemake/bin/scripts/hits-to-real-counts.py
import pandas as pd

# Read the input file specified in the Snakemake rule
df = pd.read_csv(snakemake.input[0], sep='\t')

# Check if the necessary columns exist
if 'Hits' not in df.columns or 'Count' not in df.columns:
    raise KeyError("The input file does not contain 'Hits' and/or 'Count' columns.")

# Create the 'RealCount' column if it does not exist
if 'RealCount' not in df.columns:
    df['RealCount'] = 0

# Apply the condition to create/update the 'RealCount' column
df['RealCount'] = df.apply(lambda x: x['Hits'] if x['Count'] > 0 else 0, axis=1)

# Save the updated dataframe to the output file specified in the Snakemake rule
df.to_csv(snakemake.output[0], sep='\t', index=False)
